Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -117,43 +139,57 @@ public class HTTP {
         public let url: String
     }
 
-    public static func downloadToolchain(
+    public func downloadToolchain(
+        _ toolchain: ToolchainVersion,
         url: String,
         to destination: String,
         reportProgress: @escaping (DownloadProgress) -> Void
     ) async throws {
-        let fileHandle = try FileHandle(forWritingTo: URL(fileURLWithPath: destination))
-        defer {
-            try? fileHandle.close()
-        }
+        if let downloader = self.downloader {
+            return try await downloader.downloadToolchain(
+                toolchain,
+                url: url,
+                to: destination,
+                reportProgress: reportProgress
+            )
+        } else {
+            let fileHandle = try FileHandle(forWritingTo: URL(fileURLWithPath: destination))
```

Comment:
Why is this not implemented as a ToolchainDownloader

Reply:
No good reason in particular, updated to be.